libr/util/str.c

```diff
@@ -3583,6 +3656,37 @@ R_API const char *r_str_str_xy(const char *s, const char *word, const char *prev
 	return d;
 }
 
+R_API size_t r_charset_encode_str(ut8 *asciistr, ut8 *in, size_t len_input, RCharset *r_char) {
+	bool is_in_set;
+	size_t j, k;
+	memset(asciistr, 0, len_input);
```

Comment:
unecessary memset, the buffer must be nullified only 1 byte and at the end of the loop

Reply:
It is already what `asciistr[strlen ((char *) asciistr)] = '\0';` does.

Reply:
I do not really see what you mean.

Reply:
just remove this line, its unnecessary

Reply:
Which line? memset? When I remove memset I see strange chars on pst.

Reply:
Yes the memset must be removed only the bytes needed must be written. So set the first byte to 0 and it should be gold

Reply:
I mean: good

Reply:
The strcat can not add code on non nulled char array. Else it adds after. I am trying to remove the strcat.